lib/mayaUsd/ufe/private/Utils.cpp

```diff
@@ -129,53 +129,61 @@ UsdGeomXformCommonAPI convertToCompatibleCommonAPI(const UsdPrim& prim)
-                                         prim.GetName().GetString().c_str(), 
-                                         layerDisplayNames.c_str());
-        throw std::runtime_error(err.c_str());
-    }
 
-    // if the current layer doesn't have any contributions
-    if (!MayaUsdUtils::doesEditTargetLayerContribute(prim)) {
-        auto strongestContributingLayer = MayaUsdUtils::strongestContributingLayer(prim);
-        std::string err = TfStringPrintf("Cannot %s [%s]. It is defined on another layer. Please set [%s] as the target layer to proceed.", 
-                                         commandName.c_str(),
-                                         prim.GetName().GetString().c_str(),
-                                         strongestContributingLayer->GetDisplayName().c_str());
-        throw std::runtime_error(err.c_str());
+        if(!layerDisplayName.empty())
+        {
+            layerDisplayName.pop_back();
+            std::string err = TfStringPrintf("Cannot %s [%s]. It is defined on another layer. Please set %s as the target layer to proceed.",
+                                     commandName.c_str(),
+                                     prim.GetName().GetString().c_str(), 
+                                     layerDisplayName.c_str());
+            throw std::runtime_error(err.c_str());
+        }
     }
     else
     {
-        auto layers = MayaUsdUtils::layersWithContribution(prim);
-        // if we have more than 2 layers that contributes to the final composed prim
-        if (layers.size() > 1) {
-            std::string layerDisplayNames;
-
-            // skip the the first arc which is PcpArcTypeRoot
-            // we are interested in all the arcs after root
-            std::for_each(std::next(layers.begin()),layers.end(), [&](const auto& it) {
-                 layerDisplayNames.append("[" + it->GetDisplayName() + "]" + ",");
-            });
-
-            layerDisplayNames.pop_back();
-            std::string err = TfStringPrintf("Cannot %s [%s]. It has definitions or opinions on other layers. Opinions exist in %s",
-                                             commandName.c_str(),
-                                             prim.GetName().GetString().c_str(), 
-                                             layerDisplayNames.c_str());
-            throw std::runtime_error(err.c_str());
+        for (const auto& spec : primStack) {
+            layerDisplayName.append("[" + spec->GetLayer()->GetDisplayName() + "]" + ",");
         }
+        layerDisplayName.pop_back();
+        std::string err = TfStringPrintf("Cannot %s [%s]. It is defined on another layer. Please set %s as the target layer to proceed.",
+                                 commandName.c_str(),
+                                 prim.GetName().GetString().c_str(), 
+                                 layerDisplayName.c_str());
+        throw std::runtime_error(err.c_str());
     }
 }
```

Comment:
This algorithm heavily relies on dealing with "Prim's PrimStack" rather than "Stage's LayerStack" which makes it much easier to interrogate about the SdfPrimSpec(s) in different composition arcs.